target/linux/ath79/dts/ar9344_mikrotik_routerboard-2011.dtsi

```diff
@@ -0,0 +1,209 @@
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			debounce-interval = <60>;
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+};
+
+&builtin_switch {
+	/delete-property/ qca,phy4-mii-enable; /* need to delete this to make all 5 FE ports work */
+};
+
+&eth0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	phy-mode = "rgmii-rxid";
+	pll-data = <0x6f000000 0x00000101 0x00001616>; /* default for ar934x, except for 1000M */
+	status = "okay";
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+
+	gmac-config {
+		device = <&gmac>;
+		rgmii-gmac0 = <1>;
+		rxd-delay = <1>;
+		rxdv-delay = <1>;
+	};
+
+	port@0 {
+		compatible = "swconfig,port";
+		reg = <0>;
+		swconfig,segment = "lan";
+		swconfig,portmap = <1 1>;
+	};
+
+	port@1 {
+		compatible = "swconfig,port";
+		reg = <1>;
+		swconfig,segment = "lan";
+		swconfig,portmap = <2 2>;
+	};
+
+	port@2 {
+		compatible = "swconfig,port";
+		reg = <2>;
+		swconfig,segment = "lan";
+		swconfig,portmap = <3 3>;
+	};
+
+	port@3 {
+		compatible = "swconfig,port";
+		reg = <3>;
+		swconfig,segment = "lan";
+		swconfig,portmap = <4 4>;
+	};
+
+	port@4 {
+		compatible = "swconfig,port";
+		reg = <4>;
+		swconfig,segment = "lan";
+		swconfig,portmap = <5 5>;
+	};
+
+	port@5 {
+		compatible = "swconfig,port";
+		reg = <5>;
+		swconfig,segment = "lan";
+		swconfig,portmap = <6 6>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	gmac-config {
+		device = <&gmac>;
+		switch-phy-swap = <0>;
+		switch-only-mode = <1>;
+	};
+};
+
+&mdio0 {
+	phy-mask = <0>;
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+
+		qca,ar8327-initvals = <
+			0x04 0x07c00000 /* PORT0 PAD MODE CTRL: Mac0_rgmii_en | Mac0_rgmii_txclk_delay_en | Mac0_rgmii_rxclk_delay_en | Mac0_rgmii_txclk_delay_sel3 | Mac0_rgmii_rxclk_delay_sel0 */
+			0x0c 0x00000080 /* PORT6 PAD MODE CTRL: Mac6_sgmii_en */
+			0x10 0x81000080 /* PWS_REG: POWER_ON_STRAP | LED_OPEN_EN_CSR | SERDES_AEN*/
+			0x50 0xc731c731 /* LED_CTRL0 */
+			0x54 0x00000000 /* LED_CTRL1 */
+			0x58 0x00008000 /* LED_CTRL2: 5A: disable LCD power */
+			0x5c 0x0230c300 /* LED_CTRL3: 5C: enable SFP power */
+			0x7c 0x0000007e /* PORT0_STATUS: DUPLEX_MODE_0 | RX_FLOW_EN_0 | TX_FLOW_EN_0 | RXMAC_EN_0 | TXMAC_EN_0 | SPEED_0 1000M */
+			0x94 0x0000007e /* PORT6_STATUS: DUPLEX_MODE_0 | RX_FLOW_EN_0 | TX_FLOW_EN_0 | RXMAC_EN_0 | TXMAC_EN_0 | SPEED_0 1000M */
+			0xe0 0xc701e7d0 /* SGMII_CTRL */
+		>;
+	};
+};
+
+&nand {
+	nand-ecc-mode = "soft";
+	qca,nand-scan-fixup;
+	qca,nand-swap-dma;
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "booter";
+			reg = <0x0000000 0x0040000>;
+			read-only;
+		};
+
+		partition@40000 {
+			label = "kernel";
+			reg = <0x0040000 0x03c0000>;
+		};
+
+		partition@400000 {
+			label = "ubi";
+			reg = <0x0400000 0x7c00000>;
+		};
```

Comment:
```
 		partition@40000 {
 			label = "kernel";
			reg = <0x0040000 0x07c0000>;
 		};
 
		partition@800000 {
 			label = "ubi";
			reg = <0x0800000 0x7800000>;
 		};
```
Better leave 8MB for kernel partition, kernels are growing and you need free space for bad sectors

Reply:
Done.